--- a/grid.h	2023-07-25 15:36:32
+++ b/grid.h	2023-07-25 15:40:54
@@ -1488,7 +1488,8 @@
 		case 2: return create_grid_encoding_templated_2<T, 2, HASH_TYPE>(n_dims_to_encode, encoding);
 		case 4: return create_grid_encoding_templated_2<T, 4, HASH_TYPE>(n_dims_to_encode, encoding);
 		case 8: return create_grid_encoding_templated_2<T, 8, HASH_TYPE>(n_dims_to_encode, encoding);
-		default: throw std::runtime_error{"GridEncoding: n_features_per_level must be 1, 2, 4, or 8."};
+		case 16: return create_grid_encoding_templated_2<T, 16, HASH_TYPE>(n_dims_to_encode, encoding);
+		default: throw std::runtime_error{"GridEncoding: n_features_per_level must be 1, 2, 4, 8, 16."};
 	}
 }
 
